src/lazy/encoder/mod.rs

```diff
@@ -564,4 +708,64 @@ mod tests {
         };
         writer_test(expected, test)
     }
+
+    #[test]
+    fn write_sexp() -> IonResult<()> {
```

Comment:
Not urgent, but it would be good to have a test case demonstrating that multiple levels of nested containers will work properly.